graphql_client_codegen/src/generated_module.rs

```diff
@@ -6,17 +6,23 @@ use crate::{
 use heck::*;
 use proc_macro2::{Ident, Span, TokenStream};
 use quote::quote;
-use thiserror::Error;
+use std::{error::Error, fmt::Display};
 
-#[derive(Debug, Error)]
-#[error(
-    "Could not find an operation named {} in the query document.",
-    operation_name
-)]
+#[derive(Debug)]
 struct OperationNotFound {
     operation_name: String,
 }
 
+impl Display for OperationNotFound {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("Could not find an operation named ")?;
```

Comment:
Why not `write!(f, "Could not… {} …in the query document.", self.operation_name)`?

Reply:
Definitely an unnecessary micro-optimization, but `write!()` (format macros in generals) expands to more code than it looks like. It can add up to larger binaries and compile times. I just checked on rust-playground:

```rust
use std::fmt::Write;

fn main() {
    let mut out = String::new();
    write!(out, "{} {} {}", ">", "abcd", "<");
}
```

expands to

```rust
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2018::*;
#[macro_use]
extern crate std;
use std::fmt::Write;

fn main() {
    let mut out = String::new();
    out.write_fmt(::core::fmt::Arguments::new_v1(&["", " ", " "],
                                                 &match (&">", &"abcd", &"<")
                                                      {
                                                      (arg0, arg1, arg2) =>
                                                      [::core::fmt::ArgumentV1::new(arg0,
                                                                                    ::core::fmt::Display::fmt),
                                                       ::core::fmt::ArgumentV1::new(arg1,
                                                                                    ::core::fmt::Display::fmt),
                                                       ::core::fmt::ArgumentV1::new(arg2,
                                                                                    ::core::fmt::Display::fmt)],
                                                  }));
}
```